test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -5711,5 +5711,76 @@ public void Dotnet_New_Template_Restore_Pack_Success(string template)
                 Assert.True(File.Exists(nupkgPath));
             }
         }
+
+
+        [Theory]
+        [InlineData("net472")]
+        [InlineData("net5.0")]
+        public void PackCommand_PackProject_PackageReference_PreReleaseDependency(string platform)
+        {
+            // Arrange
+            using (var testDirectory = msbuildFixture.CreateTestDirectory())
+            {
+                var dependencyName = "NuGet.Packaging";
+                var dependencyVersion = "6.0.0-preview.3";
+
+                var projectName = "ClassLibrary1";
+                var workingDirectory = Path.Combine(testDirectory, projectName);
+                var projectFile = Path.Combine(workingDirectory, $"{projectName}.csproj");
+                msbuildFixture.CreateDotnetNewProject(testDirectory, projectName);
+
+                using (var stream = new FileStream(projectFile, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    var xml = XDocument.Load(stream);
+                    ProjectFileUtils.SetTargetFrameworkForProject(xml, "TargetFramework", platform);
+                    ProjectFileUtils.AddProperty(xml, "TreatWarningsAsErrors", "true");
+                    ProjectFileUtils.AddProperty(xml, "Version", "1.2.3");
+
+                    var attributes = new Dictionary<string, string>();
+                    attributes["NoWarn"] = "NU5104";
+
+                    attributes["Version"] = dependencyVersion;
+                    ProjectFileUtils.AddItem(
+                        xml,
+                        "PackageReference",
+                        dependencyName,
+                        string.Empty,
+                        new Dictionary<string, string>(),
+                        attributes);
+
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+
+                // Act
+                msbuildFixture.RestoreProject(workingDirectory, projectName, string.Empty);
+                msbuildFixture.RunDotnet(workingDirectory, $"pack -o {workingDirectory}", false);
```

Comment:
Can you explain how this test verifies the expected behavior?  As long as this method doesn't throw, then we know the warning was suppressed?